web/src/app/services/IntegrationKeyList.js

```diff
@@ -53,134 +55,112 @@ const sortItems = (a, b) => {
   return 0
 }
 
-@withStyles(styles)
-class IntegrationKeyDetails extends React.PureComponent {
-  static propTypes = {
-    href: p.string.isRequired,
-    label: p.string.isRequired,
-    type: p.string.isRequired,
+export function IntegrationKeyDetails(props) {
+  let copyText = <CopyText title={'Copy ' + props.label} value={props.href} />
 
-    // provided by withStyles
-    classes: p.object,
+  // if link is not properly present, do not display to copy
+  if (props.type === 'email' && !props.href.startsWith('mailto:')) {
+    copyText = null
   }
 
-  state = {
-    showTooltip: false,
-  }
+  return (
+    <React.Fragment>
+      {copyText}
+      {props.type === 'email' && (
+        <RequireConfig
+          configID='Mailgun.Enable'
+          else='Email integration keys are currently disabled.'
+        />
+      )}
+    </React.Fragment>
+  )
+}
 
-  render() {
-    let tooltip = (
-      <CopyText title={'Copy ' + this.props.label} value={this.props.href} />
-    )
+export default function IntegrationKeyList(props) {
+  const classes = useStyles()
 
-    // if link is not properly present, do not display to copy
-    if (this.props.type === 'email' && !this.props.href.startsWith('mailto:')) {
-      tooltip = null
-    }
+  const [create, setCreate] = useState(false)
+  const [deleteDialog, setDeleteDialog] = useState(null)
 
-    return (
-      <React.Fragment>
-        {tooltip}
-        {this.props.type === 'email' && (
-          <RequireConfig
-            configID='Mailgun.Enable'
-            else='Email integration keys are currently disabled.'
-          />
-        )}
-      </React.Fragment>
-    )
-  }
-}
-
-@withStyles(styles)
-export default class IntegrationKeyList extends React.PureComponent {
-  static propTypes = {
-    serviceID: p.string.isRequired,
-  }
+  const { loading, error, data } = useQuery(query, {
+    variables: { serviceID: props.serviceID },
+  })
 
-  state = {
-    create: false,
-    delete: null,
+  const typeLabels = {
+    generic: 'Generic API Key',
+    grafana: 'Grafana Webhook URL',
+    site24x7: 'Site24x7 Webhook URL',
+    email: 'Email Address',
   }
-
-  render() {
-    return (
-      <React.Fragment>
-        <Grid item xs={12} className={this.props.classes.spacing}>
-          <Card>
-            <CardContent>{this.renderQuery()}</CardContent>
-          </Card>
-        </Grid>
-        <CreateFAB
-          onClick={() => this.setState({ create: true })}
-          title='Create Integration Key'
+  if (loading && !data) return <Spinner />
+  if (error) return <GenericError error={error.message} />
+
+  const items = (data.service.integrationKeys || [])
+    .slice()
+    .sort(sortItems)
+    .map((key) => ({
+      title: key.name,
+      subText: (
+        <IntegrationKeyDetails
+          key={key.id}
+          href={key.href}
+          label={typeLabels[key.type]}
+          type={key.type}
+          classes={props.classes}
```

Comment:
I don't think this is being used anymore.

Reply:
When I tried removing these lines and running a cypress test it would fail. Is it alright for me to leave them how they are?

Reply:
Just the `classes` line, the other three are used.

Reply:
Removed the `classes` line. Linter and Cypress tests passed.